pkg/admission/validator/shoot.go

```diff
@@ -83,10 +83,10 @@ func (s *shoot) Validate(ctx context.Context, new, old client.Object) error {
 		return fmt.Errorf("wrong object type %T", new)
 	}
 
-	// Get credentials
+	// Get validated credentials
 	credentials, err := openstack.GetCredentialsBySecretBinding(ctx, s.client, client.ObjectKey{Namespace: shoot.Namespace, Name: shoot.Spec.SecretBindingName})
 	if err != nil {
-		return fmt.Errorf("could not get credentials from SecretBindingName %s: %w", shoot.Spec.SecretBindingName, err)
+		return fmt.Errorf("invalid cloud credentials: %v", err)
```

Comment:
From what I see in the code, `GetCredentialsBySecretBinding` could fail not only because of invalid credentials, but also because getting the secretbinding or the secret failed. Perhaps the message could be further improved. Even better, `GetCredentialsBySecretBinding` (which is only used here) could be refactored into 2 separate calls: `GetSecretBySecretBinding`, and `ExtractCredentials`, with errors wrapped by different appropriate messages.

Reply:
Thanks for the feedback! Locally I try to apply it.

Reply:
Done, please take a look again.

Reply:
Thanks!